Filters: Remove padding and combine level with fields

matyax opened the pull request:
This PR removes the left margin from the control rows, which was initially added to give a sense of hierarchy between these filters and the indexed labels on top.

![margin](https://github.com/user-attachments/assets/19fc43d9-6e8d-434c-82b7-7c5be4e21e7e)

After making this change, it became evident that we can make a little bit of a better use of the available space, by combining the level filters with the field filters. With these changes, the level filter will not wrap, while the fields filter can freely grow.

![combined](https://github.com/user-attachments/assets/854189e6-14ae-4b30-aded-5a27bb6c0dee)

Related with #971

L2D2Grafana reviewed (approved):
lgtm, thanks for jumping right on this issue from slack! 
![Screenshot 2025-04-03 at 7 40 32 AM](https://github.com/user-attachments/assets/85a7a66a-dd7f-4524-9e70-b8390d487635)
In super small I wouldn't mind stacking them. Also seems like the patterns aren't responsive in small, perhaps a something to look into later

gtk-grafana commented:
LGTM, but agree with Liza we should put stuff on it's own row for smaller viewports
<img width="509" alt="image" src="https://github.com/user-attachments/assets/2fcf338b-d94b-4c01-ad69-086d9ac675f9" />

matyax commented:
Updated with responsiveness. Also added a breakpoint to the top header filters which were collapsing in a weird way.

https://github.com/user-attachments/assets/c67fa92a-2347-49a0-abe8-32f95c008bc2
